torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -305,6 +305,9 @@ void Reducer::initialize_buckets(std::vector<std::vector<size_t>> indices) {
       }
 
       // Allocate bucket contents tensor.
+      // This must be a Variable because as of Apr 2019 there is still
+      // a distinction between the Tensor and Variable types, and it
+      // is not recommended (or sometimes even possible) to mix and match.
       replica.contents = torch::autograd::make_variable_consuming(
           at::empty({static_cast<long>(offset)}, options));
```

Comment:
@ezyang If I didn't make this a Variable I would get casting error when trying to copy to/from it.

If there is a better way do to this, or an alternative coming up, then happy to adapt.